understory_responder/src/dispatcher.rs

```diff
@@ -129,32 +184,90 @@ use crate::types::{Dispatch, Outcome};
 /// ];
 ///
 /// let mut seen: Vec<(Phase, u32)> = Vec::new();
-/// let consumed = run(&seq, &mut (), |d, _| {
+/// let stopped = run(&seq, &mut (), |d, _| {
 ///     seen.push((d.phase, d.node.0));
 ///     if d.phase == Phase::Capture && d.node.0 == 1 { Outcome::Stop } else { Outcome::Continue }
 /// });
 ///
-/// // Not consumed; propagation aborted after the first capture.
-/// assert!(!consumed);
+/// // Propagation aborted after the first capture; we stopped at that entry.
+/// assert!(stopped.is_some());
 /// assert_eq!(seen, vec![(Phase::Capture, 1)]);
 /// ```
-pub fn run<K, W, M, E>(
-    seq: &[Dispatch<K, W, M>],
+pub fn run<'a, K, W, M, E>(
+    seq: &'a [Dispatch<K, W, M>],
     event: &mut E,
     mut handler: impl FnMut(&Dispatch<K, W, M>, &mut E) -> Outcome,
-) -> bool {
+) -> Option<&'a Dispatch<K, W, M>> {
     // The router already emits dispatch entries in capture → target → bubble
     // order, grouped by phase. We simply walk them in sequence and apply the
     // outcome rules.
     for d in seq {
         match handler(d, event) {
             Outcome::Continue => {}
             // Abort propagation immediately (spec-aligned: no target/bubble if raised in capture).
-            Outcome::Stop => return false,
-            Outcome::StopAndConsume => return true,
+            Outcome::Stop => return Some(d),
         }
     }
-    false
+    None
+}
+
+/// Event state inspection for higher‑level dispatcher helpers.
+///
+/// Implement this on your event type if you want to use
+/// [`run_with_result`]. It lets the dispatcher summarize whether the
+/// event was consumed/handled and whether any default action should be
+/// skipped, without baking those concepts into [`Outcome`].
+pub trait EventState {
+    /// Returns `true` if the event considers itself consumed/handled.
+    ///
+    /// This is a caller‑defined notion: you can map it to whatever
+    /// semantics make sense in your framework (e.g., a `handled` flag,
+    /// an a11y activation, etc.).
+    fn consumed(&self) -> bool;
+
+    /// Returns `true` if default actions should be skipped.
+    ///
+    /// This typically corresponds to a DOM‑style `default_prevented`
+    /// bit set by handlers at the target or in bubbling.
+    fn default_prevented(&self) -> bool;
+}
```

Comment:
I don't see what benefit this trait provides. In order to actually do anything with default_prevented or consumed it needs to be handled within the `handler` function. this doesn't provide any additional internal handling. It only pushes users probably implement and handle these things